modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialVariable.java

```diff
@@ -17,36 +17,52 @@
 @Data
 public class CredentialVariable {
 
-    /**
-     * The name of the CredentialVariable.
-     */
     @NotNull
     @Schema(description = "The name of the CredentialVariable,"
             + "this field is provided by the the plugin of cloud service provider.")
     private final String name;
 
-    /**
-     * The description of the CredentialVariable.
-     */
     @NotNull
     @Schema(description = "The description of the CredentialVariable,"
             + "this field is provided by the plugin of cloud service provider.")
     private final String description;
 
-    /**
-     * The value of the CredentialVariable.
-     */
+    @Schema(description = "If the variable is mandatory. If is optional then the credential "
+            + "completeness check will ignore this variable. It is upto the plugin to decide what "
+            + "needs to be done if this optional credential variable is present.",
+            defaultValue = "true")
+    private final boolean isMandatory;
+
+    private String newString;

```

Comment:
What is this field for?

Reply:
@baixinsui it was a wrong commit. Removed it now.